rocks-bin/src/project/write_new.rs

```diff
@@ -0,0 +1,301 @@
+use std::{error::Error, path::PathBuf, str::FromStr};
+
+use clap::Args;
+use eyre::{eyre, Result};
+use inquire::{
+    ui::{RenderConfig, Styled},
+    validator::Validation,
+    Confirm, Select, Text,
+};
+use itertools::Itertools;
+use spdx::LicenseId;
+use spinners::{Spinner, Spinners};
+
+use crate::utils::github_metadata::{self, RepoMetadata};
+use rocks_lib::{project::Project, rockspec::LuaDependency};
+
+// General notes and ideas:
+// - Create a subdirectory for the project, unless a path is explicitly passed.
+// - Automatically detect build type by inspecting the current repo (is there a Cargo.toml? is
+//   there something that tells us it's a lua project?).
+
+#[derive(Args)]
+pub struct NewProject {
+    /// The directory of the project.
+    #[arg(long)]
+    directory: Option<PathBuf>,
+
+    /// The project's name.
+    #[arg(long)]
+    name: Option<String>,
+
+    /// The description of the project.
+    #[arg(long)]
+    description: Option<String>,
+
+    /// The license of the project. Generic license names will be inferred.
+    #[arg(long, value_parser = clap_parse_license)]
+    license: Option<LicenseId>,
+
+    /// The maintainer of this project. Does not have to be the code author.
+    #[arg(long)]
+    maintainer: Option<String>,
+
+    /// A comma-separated list of labels to apply to this project.
+    #[arg(long, value_parser = clap_parse_list)]
+    labels: Option<std::vec::Vec<String>>, // Note: full qualified name required, see https://github.com/clap-rs/clap/issues/4626
+
+    /// A version constraint on the required Lua version for this project.
+    /// Examples: ">=5.1", "5.1"
+    #[arg(long, value_parser = clap_parse_version)]
+    lua_versions: Option<LuaDependency>,
+}
+
+fn clap_parse_license(s: &str) -> std::result::Result<LicenseId, String> {
+    match validate_license(s) {
+        Ok(Validation::Valid) => Ok(parse_license(s)),
+        Err(_) | Ok(Validation::Invalid(_)) => {
+            Err(format!("unable to identify license {s}, please try again!"))
+        }
+    }
+}
+
+fn clap_parse_version(input: &str) -> std::result::Result<LuaDependency, String> {
+    LuaDependency::from_str(format!("lua {}", input).as_str()).map_err(|err| err.to_string())
+}
+
+fn clap_parse_list(input: &str) -> std::result::Result<Vec<String>, String> {
+    if let Some((pos, char)) = input
+        .chars()
+        .find_position(|&c| c != '-' && c != '_' && c != ',' && c.is_ascii_punctuation())
+    {
+        Err(format!("Unexpected punctuation '{}' found at column {}. Lists are comma separated but names should not contain punctuation!", char, pos))
+    } else {
+        Ok(input.split(',').map(|str| str.trim().to_string()).collect())
+    }
+}
+
+fn parse_license(input: &str) -> LicenseId {
+    spdx::imprecise_license_id(input).unwrap().0
```

Comment:
issue: This `unwrap` looks suspicious.
it's a private function, so I assume checks are performed before calling this? 
If so, I would suggest appending something like `_partial` to the name or marking this function as unsafe.